api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/Enhancement.java

```diff
@@ -11,7 +11,6 @@
  * <p>
  * Methods annotated {@code @Enhancement} must define exactly one parameter of one of these types:
  * <ul>
- * <li>{@link DeclarationConfig} or {@link jakarta.enterprise.lang.model.declarations.DeclarationInfo DeclarationInfo}</li>
```

Comment:
I tried to implement this and found that it isn't exactly clear _what_ should be passed to an `@Enhancement` method that declares a parameter of type `DeclarationInfo` or `DeclarationConfig`.

Say I have this:

```java
@Enhancement
@ExactType(MyService.class)
void enhance(DeclarationConfig decl) {
    ...
}
```

How is that method supposed to be called? Once with the `MyService` class, and then once for each field in the class, and then once for each method in the class?

What if it uses `DeclarationInfo`? Should it be, in addition to above, also once for each parameter of each method in the class? Note that we have `ParameterInfo`, but don't have `ParameterConfig`.

I now think this shouldn't be allowed. It should always be either `Class{Info,Config}`, or `Field*`, or `Method*`.

Reply:
I agree this shouldn't be allowed

Reply:
+1, using `DeclarationInfo` and `DeclarationConfig` should throw an exception